### PR TITLE
Migrate service_manager to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/service/service_manager.dart [hunk 1]
```diff
@@ -44,32 +44,32 @@ class ServiceConnectionManager {
...
 
   bool get isServiceAvailable => _serviceAvailable.isCompleted;
 
-  VmServiceCapabilities _serviceCapabilities;
-  VmServiceTrafficLogger serviceTrafficLogger;
+  VmServiceCapabilities? _serviceCapabilities;
+  VmServiceTrafficLogger? serviceTrafficLogger;
 
-  Future<VmServiceCapabilities> get serviceCapabilities async {
+  Future<VmServiceCapabilities?> get serviceCapabilities async {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/service/service_manager.dart:59]: this should return non-nullable VmServiceCapabilities

File: packages/devtools_app/lib/src/service/service_manager.dart [hunk 2]
```diff
@@ -44,32 +44,32 @@ class ServiceConnectionManager {
...
   }
 
-  final _registeredServiceNotifiers = <String, ImmediateValueNotifier<bool>>{};
+  final _registeredServiceNotifiers = <String?, ImmediateValueNotifier<bool>>{};
 
-  Map<String, List<String>> get registeredMethodsForService =>
+  Map<String?, List<String?>> get registeredMethodsForService =>
       _registeredMethodsForService;
-  final Map<String, List<String>> _registeredMethodsForService = {};
+  final Map<String?, List<String?>> _registeredMethodsForService = {};
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/service/service_manager.dart:72]: make all of these types non-nullable

File: packages/devtools_app/lib/src/service/service_manager.dart [hunk 3]
```diff
@@ -79,33 +79,33 @@ class ServiceConnectionManager {
...
+  InspectorServiceBase? _inspectorService;
 
   ErrorBadgeManager get errorBadgeManager => _errorBadgeManager;
   final _errorBadgeManager = ErrorBadgeManager();
 
-  ServiceExtensionManager get serviceExtensionManager =>
+  ServiceExtensionManager? get serviceExtensionManager =>
       _serviceExtensionManager;
-  ServiceExtensionManager _serviceExtensionManager;
+  ServiceExtensionManager? _serviceExtensionManager;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/service/service_manager.dart:90]: non-nullable

File: packages/devtools_app/lib/src/service/service_manager.dart [hunk 4]
```diff
@@ -136,18 +136,17 @@ class ServiceConnectionManager {
   /// Call a service that is registered by exactly one client.
   Future<Response> callService(
     String name, {
-    String isolateId,
-    Map args,
+    String? isolateId,
+    Map? args,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/service/service_manager.dart:140]: add `<String, dynamic>` here and get rid of the type cast below

File: packages/devtools_app/lib/src/service/service_manager.dart [hunk 5]
```diff
@@ -160,15 +159,15 @@ class ServiceConnectionManager {
   }
 
   Future<void> vmServiceOpened(
-    VmServiceWrapper service, {
-    @required Future<void> onClosed,
+    VmServiceWrapper? service, {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/service/service_manager.dart:162]: non-nullable

File: packages/devtools_app/lib/src/service/service_manager.dart [hunk 6]
```diff
@@ -179,9 +178,9 @@ class ServiceConnectionManager {
     // race conditions where managers cannot listen for events soon enough.
     isolateManager.vmServiceOpened(service);
     consoleService.vmServiceOpened(service);
-    serviceExtensionManager.vmServiceOpened(service, connectedApp);
+    serviceExtensionManager!.vmServiceOpened(service, connectedApp);
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/service/service_manager.dart:181]: make serviceExtensionManager.vmServiceOpened accept a non-null connectedApp. Then unwrap it here in the method call to be consistent with timelineStreamManager.vmServiceOpened. Also, we should never be calling vmServiceOpened with a null connectedApp, so no need to accept a nullable type

File: packages/devtools_app/lib/src/service/service_manager.dart [hunk 7]
```diff
@@ -279,9 +278,9 @@ class ServiceConnectionManager {
 
     _connectedState.value = const ConnectedState(true);
 
-    final isolates = [
-      ...vm.isolates,
-      if (preferences.vmDeveloperModeEnabled.value) ...vm.systemIsolates,
+    final List<IsolateRef> isolates = [
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/service/service_manager.dart:281]: put type on RHS: `final isolates = <IsolateRef>[...`

File: packages/devtools_app/lib/src/service/service_manager.dart [hunk 8]
```diff
@@ -435,23 +434,23 @@ class ServiceConnectionManager {
...
+    if (connectedApp == null || !await connectedApp!.isFlutterApp) {
       return null;
     }
 
     final viewId = await flutterViewId;
 
     return await _callServiceExtensionOnMainIsolate(
       registrations.flutterEngineEstimateRasterCache,
-      args: <String, String>{
+      args: <String, String?>{
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/service/service_manager.dart:446]: non-nullable. optionally, remove the types completely and let them be inferred

File: packages/devtools_app/lib/src/service/service_manager.dart [hunk 9]
```diff
@@ -462,26 +461,26 @@ class ServiceConnectionManager {
       registrations.displayRefreshRate,
       args: {'viewId': viewId},
     );
-    final double fps = displayRefreshRateResponse.json['fps'];
+    final double? fps = displayRefreshRateResponse.json!['fps'];
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/service/service_manager.dart:464]: make this non-nullable, add ! to the end, and remove the ! below

File: packages/devtools_app/lib/src/service/service_manager.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/service/service_manager.dart:59]: done

File: packages/devtools_app/lib/src/service/service_manager.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/service/service_manager.dart:72]: done

File: packages/devtools_app/lib/src/service/service_manager.dart (hunk 3 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/service/service_manager.dart:90]: done

File: packages/devtools_app/lib/src/service/service_manager.dart (hunk 4 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/service/service_manager.dart:140]: done

File: packages/devtools_app/lib/src/service/service_manager.dart (hunk 5 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/service/service_manager.dart:162]: done

File: packages/devtools_app/lib/src/service/service_manager.dart (hunk 6 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/service/service_manager.dart:181]: done. 
Noted serviceExtensionManager.vmServiceOpened  should not accept null. It is not null safe yet.

File: packages/devtools_app/lib/src/service/service_manager.dart (hunk 7 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/service/service_manager.dart:281]: done

File: packages/devtools_app/lib/src/service/service_manager.dart (hunk 8 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/service/service_manager.dart:446]: done

File: packages/devtools_app/lib/src/service/service_manager.dart (hunk 9 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/service/service_manager.dart:464]: done

File: packages/devtools_app/lib/src/service/service_manager.dart [hunk 10]
```diff
@@ -56,20 +56,20 @@ class ServiceConnectionManager {
...
     if (_serviceCapabilities == null) {
       await _serviceAvailable.future;
       final version = await service!.getVersion();
       _serviceCapabilities = VmServiceCapabilities(version);
     }
-    return _serviceCapabilities;
+    return _serviceCapabilities!;
   }
 
   final _registeredServiceNotifiers = <String?, ImmediateValueNotifier<bool>>{};
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/service/service_manager.dart:68]: make string non nullable here too

**Review by kenzieschmoll — APPROVED** [review]
lgtm with one comment

File: packages/devtools_app/lib/src/service/service_manager.dart (hunk 10 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/service/service_manager.dart]: done

**Comment by polina-c** [comment]
tested

**Pull request merged**